### ISSUE TITLE
Portable library for Windows Runtime ?

### ISSUE THREAD

**Issue by mpapillon** [mbdavid/LiteDB#26]
Hello,
I would like to know if you have planned to release a portable class library for Windows Phone and Windows 8. Thanks in advance for your reply.

**Comment by mbdavid** [comment]
To be honest with you, I never works with Windows Phone or Windows 8 RT. But this is not the first request about that. I'm waiting VS2015 be released to do some tests to know if is compatible and what must be changed.
I don't know, for example, if Windows Phone or WinRT has all methods in classes that I use, or I will need to re-write some code.

**Comment by dev-nextprogress** [comment]
Hi guys,
I have just created a new (local) version of LiteDB with additions that currently seem to work on Windows Phone 8 Silverlight, Android and iOS. It's not a pure PCL but linked platform projects with a few PCL helpers/extensions for things like NameValueCollection and Unicode Normalization.
The biggest challenge is that iOS does not support Reflection.Emit, so I tried to replace it with Activator-based code but I have to admit that I am not a reflection expert. So far the tests I have done seem to work. I also had to change all the Lock/Unlock logic and used a Mutex-based implementation. The FileStream locking did not work reliably in all the different .NET flavors.
Unfortunately I don't have the time right now to work with you on pulling/merging here on github. I actually didn't intend this to become such a large change, but if you are interested I will be glad to send you the code. Just e-mail me at dev@nextprogress.com.
Awesome work mbdavid. Many hanks
Hjalmar
